xrpl/wallet.go

```diff
@@ -0,0 +1,170 @@
+package xrpl
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strings"
+
+	addresscodec "github.com/Peersyst/xrpl-go/address-codec"
+	binarycodec "github.com/Peersyst/xrpl-go/binary-codec"
+	"github.com/Peersyst/xrpl-go/keypairs"
+	"github.com/Peersyst/xrpl-go/xrpl/hash"
+	"github.com/tyler-smith/go-bip32"
+	"github.com/tyler-smith/go-bip39"
+)
+
+// A utility for deriving a wallet composed of a keypair (publicKey/privateKey).
+// A wallet can be derived from either a seed, mnemonic, or entropy (array of random numbers).
+// It provides functionality to sign/verify transactions offline.
+type Wallet struct {
+	PublicKey      string
+	PrivateKey     string
+	ClassicAddress string
+	Seed           string
+}
+
+// Creates a new random Wallet. In order to make this a valid account on ledger, you must
+// Send XRP to it.
+func NewWallet(alg addresscodec.CryptoAlgorithm) (Wallet, error) {
```

Comment:
Would it make more sense to call it `Generate` like in xrpl.js ?

I am new to Go but I would expect we would import the `Wallet` and then write something like:

```go
import ( "wallet" )

wallet.Generate()
```

or also (based on your example file):

```go
import ( "xrpl" )

xrpl.Wallet.Generate()
```

Reply:
The issue with Go is that exports behave different than in javascript. Every function and struct lives in a package. So, in order to use a function (p.e.) that is stored in "x" package, it needs to be called with:

```go
import (
   "x"
)

func main() {
   x.Method()
}
```

Reply:
we could make a package `wallet` and store everything in there, but calling the methods can be weird.

```go
import (
   "wallet"
)

func main() {
   wallet, err := wallet.GenerateWallet(...)
}
```